File: problems/0801 - 1200/1137/1137.cpp
class Solution {
public:
    int tribonacci(int n) {
        if (n < 2) {
            return n;
        }
        if (n == 2) {
            return 1;
        }
        int trib_left = 0;
        int trib_mid = 1;
        int trib_right = 1;
        for (int i = 3; i <= n; ++i) {
            int tmp = trib_left + trib_mid + trib_right;
            trib_left = trib_mid;
            trib_mid = trib_right;
            trib_right = tmp;
        }
        return trib_right;
    }
};

File: problems/0801 - 1200/1137/1137.py
class Solution:
    def tribonacci(self, n: int) -> int:
        if n < 2:
            return n
        if n == 2:
            return 1

        trib_left = 0
        trib_mid = 1
        trib_right = 1
        for _ in range(3, n + 1):
            trib_left, trib_mid, trib_right = trib_mid, trib_right, trib_left + trib_mid + trib_right
        return trib_right
